src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -293,7 +293,7 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
             }
 
             if (!$type) {
-                if (isset($arguments[$index])) {
+                if (\array_key_exists($index, $arguments)) {
```

Comment:
This cannot be the correct patch: there is no reason why the concern applies only when there is no type.

Instead, I cannot recall why I added the "if" on L290 just above. This if means that `''` and `null` are considered as wildcard for letting autowiring process even when an attribute is used. Removing it looks like the right fix to me.

```
--- a/src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php
+++ b/src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php
@@ -276,23 +276,17 @@ class AutowirePass extends AbstractRecursivePass
                 continue;
             }
 
-            $type = ProxyHelper::exportType($parameter, true);
-
             if ($checkAttributes) {
                 foreach ($parameter->getAttributes() as $attribute) {
                     if (\in_array($attribute->getName(), [TaggedIterator::class, TaggedLocator::class, Autowire::class, MapDecorated::class], true)) {
                         $arguments[$index] = $this->processAttribute($attribute->newInstance(), $parameter->allowsNull());
 
-                        break;
+                        continue 2;
                     }
                 }
-
-                if ('' !== ($arguments[$index] ?? '')) {
-                    continue;
-                }
             }
 
-            if (!$type) {
+            if (!$type = ProxyHelper::exportType($parameter, true)) {
                 if (isset($arguments[$index])) {
                     continue;
                 }
```

Reply:
I applied your diff, thanks for explanations